storage/remote/queue_manager.go

```diff
@@ -1642,9 +1700,14 @@ func populateMinimizedTimeSeriesStr(symbolTable *rwSymbolTable, batch []timeSeri
 		case tFloatHistogram:
 			pendingData[nPending].Histograms = append(pendingData[nPending].Histograms, FloatHistogramToMinHistogramProto(d.timestamp, d.floatHistogram))
 			nPendingHistograms++
+		case tMetadata:
+			pendingData[nPending].Metadata.Type = metricTypeToMetricTypeProtoV2(d.metadata.Type)
+			pendingData[nPending].Metadata.HelpRef = symbolTable.RefStr(d.metadata.Help)
+			pendingData[nPending].Metadata.HelpRef = symbolTable.RefStr(d.metadata.Unit)
+			nPendingMetadata++
 		}
```

Comment:
Interesting. I thought metadata was going to be added to every timeseries instance, not that it was going to be sent separately. I suppose it is done this way because there's no guarantee a series' sample and its metadata will be encountered together while building a single request?

If we are going to do this, we will want to make the `Metadata` field in the proto `optional`. No?

Reply:
+1

I believe the approach on the [last PR](https://github.com/prometheus/prometheus/pull/11640/files#diff-1e66d5e658a9d99e1bd9b7be7d626d6b3bf9bbf480e210d2946c562cd746767cR1486) was to take the metadata added on each of the Append/AppendExemplar/AppendHistograms call and if it's to be sent, then record it along all series. Even sending an empty record at the time would be OK I believe.

Reply:
Nope, you're both right. This is a mistake :+1: